sklearn/preprocessing/tests/test_discretization.py

```diff
@@ -174,3 +175,38 @@ def test_numeric_stability():
         X = X_init / 10**i
         Xt = KBinsDiscretizer(n_bins=2).fit_transform(X)
         assert_array_equal(Xt_expected, Xt)
+
+
+def test_encode():
+    # test invalid encode option
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3], encode='invalid-encode')
+    assert_raises(ValueError, est.fit, X)
+
+    # test encode options through comparison
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3],
+                           encode='ordinal').fit(X)
+    expected1 = est.transform(X)
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3],
+                           encode='onehot-dense').fit(X)
+    expected2 = est.transform(X)
+    assert_array_equal(OneHotEncoder(n_values=[2, 3, 3, 3], sparse=False)
+                       .fit_transform(expected1),
+                       expected2)
+    assert_raises(ValueError, est.inverse_transform, X)
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3],
+                           encode='onehot').fit(X)
+    expected3 = est.transform(X)
```

Comment:
Should you not check that the output is sparse in the onehot case?

Also probably check that the output is not sparse in the onehot-dense case.